### PR TITLE
Fixes #11672, #11727: Delete default tab text and add padding to infinite character

### LINKED ISSUE #11672: FNX-12648 ⁃ [Bug]Tabs tray: Tapping the New Tab FAB displays ":)" instead of the tabs number
## Prerequisites
Have at least a tab opened

## Steps to reproduce
1. From the page view tap the tabs counter
1. Tapping the New tab FAB
1. Check the Tabs Counter icon 

### Expected behavior
The Tabs Counter should display the number of open tabs

### Actual behavior
Displays ":)" while the making the transition to Fenix's home screen
image

### Device information
* Android device: OnePlus A3 (Android 6.0.1)
* Fenix version: 
• Nightly 200617 from 6/17
• Beta 5.2.0-beta.2 from 6/16

### Notes
► Video

### PR DESCRIPTION
**Pull request by hakkikaancaliskan**
Fixes #11672, #11727, #11739

For #11727: 
resim

For #11672:
ezgif com-video-to-gif

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt [hunk 1]
```diff
@@ -198,6 +178,7 @@ open class TabCounter @JvmOverloads constructor(
 
     private fun formatForDisplay(count: Int): String {
         return if (count > MAX_VISIBLE_TABS) {
+            counter_text.setPadding(0,0,0,6)
```
Comment (BranescuMihai) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt:181]: Please make `6` a private const so the checkers will pass

File: app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt [hunk 2]
```diff
@@ -217,14 +198,14 @@ open class TabCounter @JvmOverloads constructor(
     }
 
     companion object {
+        var INTERNAL_COUNT = 0
```
Comment (BranescuMihai) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt:201]: Can this be an instance variable like it was before or is being static required? If it's required, then please make it internal.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt [hunk 3]
```diff
@@ -48,29 +43,15 @@ open class TabCounter @JvmOverloads constructor(
...
 
     fun setCountWithAnimation(count: Int) {
-        updateContentDescription(count)
-
-        // Don't animate from initial state.
-        if (this.count == 0) {
-            setCount(count)
-            return
-        }
+        setCount(count)
```
Comment (BranescuMihai) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt:46]: Since we're setting the count(so implicitly INTERNAL_COUNT) here, the following `when` case `INTERNAL_COUNT == count` will always be true, so there will never be an animation

File: app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt (hunk 2 above)
Comment (hakkikaancaliskan) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt:201]: It's required because while navigating it got reset to 0.
Gonna make it internal.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt (hunk 3 above)
Comment (hakkikaancaliskan) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt:46]: You're right! I'm going to remove it.

File: app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt [hunk 4]
```diff
@@ -48,29 +43,14 @@ open class TabCounter @JvmOverloads constructor(
...
```
Comment (BranescuMihai) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt:46]: We can't fully remove the INTERNAL_COUNT == count case because now it's the opposite scenario, we animate almost always (including when we just press '+' from tab tray and go to home fragment, even though the number stays the same). 
I think we need something along the lines of: 
`when {`
`count == INTERNAL_COUNT -> return`
            `count == 0 -> {`
                `setCount(count)`
                `return`
            `} // Initial state.`
            `count > MAX_VISIBLE_TABS -> {`
                `setCount(count)`
                `return`
            `} // There are still over MAX_VISIBLE_TABS tabs open.`
        `}`

In other words, basically we need to do nothing in case it's equal, and to do also `setCount` in the other 2 cases. @hakkikaancaliskan What do you think?

Comment (hakkikaancaliskan) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt:46]: You know the best, I'm amateur. Is there something else with pr?

Comment (hakkikaancaliskan) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt:46]: When I look to the [current master](https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt) maybe we can take setCount to bottom of when.
```
when {
            count == INTERNAL_COUNT -> return // There isn't any tab added or removed.
            count == 0 -> return // Initial state.
            count > MAX_VISIBLE_TABS -> return // There are still over MAX_VISIBLE_TABS tabs open.
        }

        setCount(count)
```

What you think @BranescuMihai?

Comment (hakkikaancaliskan) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt:46]: Sorry, my bad.

Comment (hakkikaancaliskan) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt:46]: You know the best, I'm amateur. Fixing now as you said.

Comment (hakkikaancaliskan) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt:46]: Also I'm taking `setCount` to the bottom of `when`.
What do you think?

Comment (hakkikaancaliskan) [reply app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt:46]: resim
How is this? I've tested and working great.

```
// No need to animate on these cases.
        when {
            count == INTERNAL_COUNT -> {
                Log.d("kaan", "count == INTERNAL_COUNT")
                return
            } // There isn't any tab added or removed.
            INTERNAL_COUNT == 0 -> {
                Log.d("kaan", "INTERNAL_COUNT == 0")
                setCount(count)
                return
            } // Initial state.
            count > MAX_VISIBLE_TABS && INTERNAL_COUNT > MAX_VISIBLE_TABS -> {
                Log.d("kaan", "count > MAX_VISIBLE_TABS && INTERNAL_COUNT > MAX_VISIBLE_TABS")
                INTERNAL_COUNT = count
                return
            } // There are still over MAX_VISIBLE_TABS tabs open.
        }

        Log.d("kaan", "animate")
        setCount(count)
```

**Comment by hakkikaancaliskan** [comment]
Oh no, fucked up pr :(

**Comment by hakkikaancaliskan** [comment]
@BranescuMihai really sorry for this commit disaster :(

**Comment by hakkikaancaliskan** [comment]
@BranescuMihai I've noticed bug. When I press a top site `setCountWithAnimation()` not starting, looks like it's using something different thus not animating. Do you know where and how it changes tab counter?

**Comment by hakkikaancaliskan** [comment]
Current bugs found out:
1- Have 0 tabs, start fenix, tap to a top site
actual: not animating on starting top site
expected: should animate

2- have at least 3 tab, on tab tray tap close all taps
actual: updating tab counter like 3->2->1->0 with animation
expected: update tab counter 3->0 with (or without because this isn't seen) animation

**Comment by hakkikaancaliskan** [comment]
ohh this git driving me crazy
I will open a new pr if i can't fix this disaster

**Comment by BranescuMihai** [comment]
Thanks for the fix @hakkikaancaliskan, looks good! Re-opened to run the tests (#11878)
